public/app/features/scenes/components/layout/SceneGridLayout.tsx

```diff
@@ -0,0 +1,438 @@
+                  Disable draggable if mobile device, solving an issue with unintentionally
+                  moving panels. https://github.com/grafana/grafana/issues/18497
+                  theme.breakpoints.md = 769
+                */
+              isDraggable={width > 768}
+              isResizable={false}
+              containerPadding={[0, 0]}
+              useCSSTransforms={false}
+              margin={[GRID_CELL_VMARGIN, GRID_CELL_VMARGIN]}
+              cols={GRID_COLUMN_COUNT}
+              rowHeight={GRID_CELL_HEIGHT}
+              draggableHandle={`.grid-drag-handle-${model.state.key}`}
+              layout={width > 768 ? layout.lg : layout.sm}
+              onDragStop={model.onDragStop}
+              onResizeStop={model.onResizeStop}
+              onLayoutChange={model.onLayoutChange}
+              isBounded={true}
+            >
+              {children.map((child) => {
+                /* eslint-disable-next-line */
+                const childProps: SceneGridCellRendererProps<any> = {
+                  model: child,
+                  isLayoutDraggable: width > 768,
+                };
+
+                return (
+                  <div key={child.state.key}>
+                    <child.Component {...childProps} key={child.state.key} />
+                  </div>
+                );
+              })}
+            </ReactGridLayout>
+          </div>
+        );
+      }}
+    </AutoSizer>
+  );
+}
+
+interface SceneGridCellState extends Omit<SceneLayoutState, 'size'> {
+  isResizable?: boolean;
+  isDraggable?: boolean;
+  size: GridCellLayout;
+}
+
+export class SceneGridCell extends SceneObjectBase<SceneGridCellState> {
+  static Component = SceneGridCellRenderer;
+  getLayout = (): SceneGridLayout => {
+    if (this.parent instanceof SceneGridLayout) {
+      return this.parent;
+    }
+    throw new Error('SceneGridCell must be a child of SceneGridLayout');
+  };
+}
+
+interface SceneGridCellRendererProps<T> extends SceneComponentProps<T> {
+  isLayoutDraggable?: boolean;
+}
+
+function SceneGridCellRenderer(props: SceneGridCellRendererProps<SceneGridCell>) {
+  const state = props.model.useState();
+  const isDraggable = Boolean(props.isLayoutDraggable) ? state.isDraggable : false;
+  return (
+    <div style={{ width: '100%', height: '100%', position: 'relative', zIndex: 0 }}>
+      {/* TODO: This is a temporary solution to make the grid cell draggable*/}
+      {isDraggable && props.model.getLayout().getDragHandle()}
+      <>
+        {props.model.state.children.map((child) => {
+          return <child.Component key={child.state.key} model={child} />;
+        })}
+      </>
+    </div>
+  );
+}
+
+function SceneGridDragHandle({ className }: { className?: string }) {
+  return (
+    <div
+      className={className}
+      style={{
+        width: '20px',
+        height: '20px',
+        position: 'absolute',
+        top: '5px',
+        right: '5px',
+        zIndex: 1,
+        cursor: 'move',
```

Comment:
Yes, I had a hard time with this one - this is a quick way to get things working. The main thing is that in order to support multiple grids, the Grid layout (and in a consequence the handle itself) needs to have a unique class name for the handle: `draggableHandle={`.grid-drag-handle-${model.state.key}`}` . So I wanted the layout to return a preconfigured handle element, so that the consuming component doesn't have to care about anything. I expect the handle to be an Icon with correct class name applied only, so that PanelChrome or whatever can style/position it as pleased. But again, this part is just to enable the dnd as soon as now.

Reply:
This is now moved outside of the Grid layout completely. I've added `getLayout` method to the scene object that will resolve the enclosing layout object, and then the respective object can i.e render the drag handle(or sth else). 

It's now used in the SceneGridRow and VizPanel if you wanna have a look at how the drag handle is added there.